src/vse_sync_pp/sequence.py

```diff
@@ -0,0 +1,175 @@
+            # pylint: disable=consider-using-with
+            file = stdin if source == '-' else open(source, encoding=encoding)
+            if contains == 'muxed':
+                yield muxed(file, parsers)
+            else:
+                yield fixed(file, parsers[contains])
+
+# tuple of the most recent values generated by source
+# timestamp must be first and numeric
+Head = namedtuple('Head', ('timestamp', 'id_', 'data', 'source'))
+
+def build_head(source):
+    """Return a :class:`Head` value or None from the next item in `source`."""
+    try:
+        (id_, data) = next(source)
+        return Head(data.timestamp, id_, data, source)
+    except StopIteration:
+        return None
+
+def build_heads(sources):
+    """Return a list of :class:`Head` values from the first items in `sources`.
+
+    The returned list is sorted by timestamp in ascending order.
+    """
+    heads = []
+    for source in sources:
+        head = build_head(source)
+        if head:
+            heads.append(head)
+    return sorted(heads)
+
+def insert_head(heads, head):
+    """Return `heads` with `head` inserted in ascending timestamp order.
+
+    If `head` is None then do not insert `head`.
+    """
+    if head is None:
+        return heads
+    idx = 0
+    while idx < len(heads):
+        if head.timestamp < heads[idx].timestamp:
+            break
+        idx += 1
+    heads.insert(idx, head)
+    return heads
+
+def main():
+    """Sequence log messages from multiple sources to stdout.
+
+    Log messages are read from files specified in the YAML file supplied in
+    command line argument `sources`. This YAML file may contain multiple
+    documents, with each document containing a single object with 'source' and
+    'contains' pairs. 'source' specifies a file to read from or '-' to read from
+    stdin; 'contains' specifies the content in the file, either the id of a
+    supported parser (as listed in command line help for options `--include`
+    and `--exclude`) or 'muxed' for multiplexed content.
+
+    Each line in multiplexed content must be a JSON-encoded object with 'id' and
+    'data' pairs. 'id' must be the id of a supported parser; 'data' a value
+    conforming to the canonical form produced by that parser.
+
+    The output written to stdout is multiplexed content, with lines sequenced as
+    follows. An initial set of candidate messages to write is constructed by
+    parsing the first log message from each source. The message with the lowest
+    timestamp in this set is written to stdout before being replaced by the next
+    log message from its source. This process is repeated until all sources are
+    empty.
+    """
+    aparser = ArgumentParser(description=main.__doc__)
+    aparser.add_argument(
+        '--include', choices=PARSERS, nargs='*', default=(),
+        help='restrict output messages to these parsers',
+    )
+    aparser.add_argument(
+        '--exclude', choices=PARSERS, nargs='*', default=(),
```

Comment:
working good 

```
cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml
{"id": "gnss/time-error", "data": [1111.222, 4, 0]}
{"id": "gnss/time-error", "data": [1112.222, 5, 0]}
{"id": "gnss/time-error", "data": [1113.222, 6, 0]}
{"id": "dpll/time-error", "data": [1234.567, 3, 3, 77.0]}
{"id": "dpll/time-error", "data": [1235.567, 4, 4, 78.0]}
{"id": "gnss/time-error", "data": [1333.444, 7, 0]}
{"id": "gnss/time-error", "data": [1334.444, 8, 0]}
{"id": "gnss/time-error", "data": [1334.444, 9, 0]}
{"id": "gnss/time-error", "data": [2345.444, 1, 0]}
{"id": "dpll/time-error", "data": [2345.671, 9, 9, 56.0]}
{"id": "gnss/time-error", "data": [2346.444, 2, 0]}
{"id": "gnss/time-error", "data": [2347.444, 3, 0]}
{"id": "gnss/time-error", "data": [2348.444, 4, 0]}
{"id": "dpll/time-error", "data": [3456.712, 7, 7, 6.0]}
{"id": "gnss/time-error", "data": [3457.444, 5, 0]}
{"id": "gnss/time-error", "data": [3458.444, 6, 0]}

```
```
cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml --exclude gnss/time-error
{"id": "dpll/time-error", "data": [1234.567, 3, 3, 77.0]}
{"id": "dpll/time-error", "data": [1235.567, 4, 4, 78.0]}
{"id": "dpll/time-error", "data": [2345.671, 9, 9, 56.0]}
{"id": "dpll/time-error", "data": [3456.712, 7, 7, 6.0]}
```
returns empty string (which is what we want):
` cat bar | env PYTHONPATH=src python3 -m vse_sync_pp.sequence sources.yml --exclude gnss/time-error dpll/time-error
`